lib/node_modules/@stdlib/math/base/special/normhermite-polynomial/test/test.js

```diff
@@ -32,4 +58,353 @@ tape( 'main export is a function', function test( t ) {
+		v = normhermitepoly( 1, x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 3.0 * EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `He2(x)` for positive medium numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = mediumPositive2.x;
+	expected = mediumPositive2.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = normhermitepoly( 2, x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 3.0 * EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `He5(x)` for positive medium numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = mediumPositive5.x;
+	expected = mediumPositive5.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = normhermitepoly( 5, x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 3.0 * EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `He1(x)` for negative medium numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = mediumNegative1.x;
+	expected = mediumNegative1.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = normhermitepoly( 1, x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 3.0 * EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `He2(x)` for negative medium numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = mediumNegative2.x;
+	expected = mediumNegative2.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = normhermitepoly( 2, x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 3.0 * EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `He5(x)` for negative medium numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = mediumNegative5.x;
+	expected = mediumNegative5.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = normhermitepoly( 5, x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 3.0 * EPS * abs( expected[ i ] );
```

Comment:
@jrreed83 Out of curiosity, was `3.0` determined by trial-and-error? Given that the algo only involves multiplications and is relatively straightforward, I am a bit surprised that we disagree with the reference implementation by more than `EPS`.

Reply:
You're totally right.  I just changed the tolerance to ```EPS``` and all the tests pass.  I'm not currently using a relative error either, I'm just checking the absolute error between the the python and javascript implementations.  The ```3.0``` came from the ```expm1rel``` tests.

Reply:
Unless the return values are approximately `1.0` OR you need to accommodate values which may not precisely equal 0, you should **always** use relative error.

Reply:
You can quickly test for yourself why this is the case:

```javascript
> 10.0 + EPS
10.0
```

What we are really after in using the relative error is how many ULP actual and expected values differ. When you use absolute error, you are not accounting for the fact that the delta between two consecutive floating-point numbers scales with magnitude.

Reply:
Most likely, the reason why all tests pass is that the delta between the two implementations is zero. However, as tests are documentation, we should acknowledge the reality of floating-point format. If you want to explicitly account for equality, you could do

```javascript
if ( actual[ i ] === expected[ i ] ) {
    t.equal( actual[ i ], expected[ i ], 'returns expected value. n: '+n+'. x: '+x );
} else {
    // Perform relative error comparison...
}
```